pal/samples/esp8266/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.ino

```diff
@@ -0,0 +1,256 @@
+// Note2: To use this sample with the esp32, you MUST build the AzureIoTSocket_WiFi library by using the make_sdk.py,
+//        found in https://github.com/Azure/azure-iot-pal-arduino/tree/master/build_all. 
+//        Command line example: python3 make_sdk.py -o <your-lib-folder>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+// You must set the device id, device key, IoT Hub name and IotHub suffix in
+// iot_configs.h
+#include "iot_configs.h"
+#include "sample_init.h"
+#ifdef is_esp_board
+  #include "Esp.h"
+#endif
+
+
+static char ssid[] = IOT_CONFIG_WIFI_SSID;
+static char pass[] = IOT_CONFIG_WIFI_PASSWORD;
+
+#ifdef SAMPLE_MQTT
+    #include "AzureIoTProtocol_MQTT.h"
+    #include "iothubtransportmqtt.h"
+#endif // SAMPLE_MQTT
+#ifdef SAMPLE_HTTP
+    #include "AzureIoTProtocol_HTTP.h"
+    #include "iothubtransporthttp.h"
+#endif // SAMPLE_HTTP
+
+/* Define several constants/global variables */
+static const char* connectionString = DEVICE_CONNECTION_STRING;
+#define MESSAGE_COUNT 5 // determines the number of times the device tries to send a message to the IoT Hub in the cloud.
+static bool g_continueRunning = true; // defines whether or not the device maintains its IoT Hub connection after sending (think receiving messages from the cloud)
+static size_t g_message_count_send_confirmations = 0;
+
+IOTHUB_MESSAGE_HANDLE message_handle;
+size_t messages_sent = 0;
+const char* telemetry_msg = "test_message";
+
+// Select the Protocol to use with the connection
+#ifdef SAMPLE_MQTT
+    IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol = MQTT_Protocol;
+#endif // SAMPLE_MQTT
+#ifdef SAMPLE_HTTP
+   IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol = HTTP_Protocol;
+#endif // SAMPLE_HTTP
+
+IOTHUB_DEVICE_CLIENT_LL_HANDLE device_ll_handle;
+
+static int callbackCounter;
+int receiveContext = 0;
+
+/* -- receive_message_callback --
+ * Callback method which executes upon receipt of a message originating from the IoT Hub in the cloud. 
+ * Note: Modifying the contents of this method allows one to command the device from the cloud. 
+ */
+static IOTHUBMESSAGE_DISPOSITION_RESULT receive_message_callback(IOTHUB_MESSAGE_HANDLE message, void* userContextCallback)
+{
+    int* counter = (int*)userContextCallback;
+    const unsigned char* buffer;
+    size_t size;
+    const char* messageId;
+
+    // Message properties
+    if ((messageId = IoTHubMessage_GetMessageId(message)) == NULL)
+    {
+        messageId = "<null>";
+    }
+
+    // Message content
+    if (IoTHubMessage_GetByteArray(message, (const unsigned char**)&buffer, &size) != IOTHUB_MESSAGE_OK)
+    {
+        LogInfo("unable to retrieve the message data\r\n");
+    }
+    else
+    {
+        LogInfo("Received Message [%d]\r\n Message ID: %s\r\n Data: <<<%.*s>>> & Size=%d\r\n", *counter, messageId, (int)size, buffer, (int)size);
+        // If we receive the work 'quit' then we stop running
+        if (size == (strlen("quit") * sizeof(char)) && memcmp(buffer, "quit", size) == 0)
```

Comment:
>strlen("quit") [](start = 21, length = 14)

Define constant for this literal.

To get the size of literal use sizeof(LITERAL) - 1